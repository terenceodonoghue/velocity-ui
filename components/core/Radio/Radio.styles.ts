import { css, FunctionInterpolation, Theme } from '@emotion/react';

export const radioButton: FunctionInterpolation<Theme> = ({ palette }) => css`
  display: inline-block;
  border: solid 1px rgba(128, 151, 177, 0.5);
  border-radius: 50%;
  height: 16px;
  margin: 0 8px 0 0;
  width: 16px;

  input:disabled + & {
    opacity: 0.4;
  }

  input:checked:enabled + & {
    border: solid 4px ${palette.accent};
  }
`;

export const radioLabel: FunctionInterpolation<Theme> = ({ typography }) => css`
  display: flex;
  align-items: center;
  font-weight: ${typography.fontWeight.medium};
  cursor: pointer;

  input {
    height: 0;
    margin: 0;
    width: 0;
    opacity: 0;
  }
`;
