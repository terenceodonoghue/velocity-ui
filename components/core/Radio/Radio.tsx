import React, { forwardRef, InputHTMLAttributes } from 'react';
import * as css from './Radio.styles';

const Radio = forwardRef<
  HTMLInputElement,
  InputHTMLAttributes<HTMLInputElement>
>(({ children, ...props }, ref) => (
  <label css={css.radioLabel} data-testid="radio">
    <input data-testid="radio-input" ref={ref} type="radio" {...props} />
    <span css={css.radioButton} />
    {children}
  </label>
));

export default Radio;
