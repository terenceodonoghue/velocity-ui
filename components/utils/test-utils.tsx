import React, { FunctionComponent, ReactElement } from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import ThemeProvider from '~/components/core/ThemeProvider';

const Providers: FunctionComponent = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
): RenderResult => render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

export { customRender as render };
