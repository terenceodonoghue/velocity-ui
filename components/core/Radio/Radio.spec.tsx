import React from 'react';
import faker from 'faker';
import { render, userEvent } from 'test-utils';
import Radio from './Radio';

describe('Radio', () => {
  it('renders label', () => {
    const content = faker.hacker.phrase();
    const { container, getByTestId } = render(<Radio>{content}</Radio>);

    expect(getByTestId('radio')).toHaveTextContent(content);
    expect(container).toMatchSnapshot();
  });

  it('is unchecked by default', () => {
    const { getByTestId } = render(<Radio />);

    expect(getByTestId('radio-input')).not.toBeChecked();
  });

  it('is checked when clicked', () => {
    const { getByTestId } = render(<Radio />);

    userEvent.click(getByTestId('radio'));

    expect(getByTestId('radio-input')).toBeChecked();
  });
});
