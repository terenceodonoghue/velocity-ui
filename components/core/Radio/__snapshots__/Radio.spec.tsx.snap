// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio renders label 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 500;
  cursor: pointer;
}

.emotion-0 input {
  height: 0;
  margin: 0;
  width: 0;
  opacity: 0;
}

.emotion-1 {
  display: inline-block;
  border: solid 1px rgba(128, 151, 177, 0.5);
  border-radius: 50%;
  height: 16px;
  margin: 0 8px 0 0;
  width: 16px;
}

input:disabled+.emotion-1 {
  opacity: 0.4;
}

input:checked:enabled+.emotion-1 {
  border: solid 4px #2e5bff;
}

<div>
  <label
    class="emotion-0"
    data-testid="radio"
  >
    <input
      data-testid="radio-input"
      type="radio"
    />
    <span
      class="emotion-1"
    />
    The SMS program is down, synthesize the redundant feed so we can synthesize the SQL hard drive!
  </label>
</div>
`;
